projection/src/test/scala/akka/projection/dynamodb/DynamoDBTimestampOffsetProjectionSpec.scala

```diff
@@ -1065,6 +1099,326 @@ class DynamoDBTimestampOffsetProjectionSpec
+
+      val startTime = TestClock.nowMicros().instant()
+      val envelopes1 = createEnvelopesUnknownSequenceNumbers(startTime, pid1, pid2)
+      val sourceProvider1 = createBacktrackingSourceProvider(envelopes1)
+      implicit val offsetStore = createOffsetStore(projectionId, sourceProvider1)
+
+      val handlerProbe = createTestProbe[String]()
+
+      val projection1 =
+        DynamoDBProjection
+          .exactlyOnceGroupedWithin(
+            projectionId,
+            Some(settings),
+            sourceProvider1,
+            handler = () => new TransactGroupedConcatHandler(handlerProbe.ref))
+          .withGroup(3, 3.seconds)
+
+      projectionTestKit.run(projection1) {
+        projectedTestValueShouldBe("e1-1|e1-2|e1-3")(pid1)
+        projectedTestValueShouldBe("e2-1")(pid2)
+      }
+      latestOffsetShouldBe(envelopes1.collectFirst { case env if env.event == "e1-3" => env.offset }.get)
+
+      // simulate backtracking
+      logger.debug("Starting backtracking")
+      val envelopes2 = createEnvelopesBacktrackingUnknownSequenceNumbers(startTime, pid1, pid2)
+      val sourceProvider2 = createBacktrackingSourceProvider(envelopes2)
+      val projection2 =
+        DynamoDBProjection
+          .exactlyOnceGroupedWithin(
+            projectionId,
+            Some(settings),
+            sourceProvider2,
+            handler = () => new TransactGroupedConcatHandler(handlerProbe.ref))
+          .withGroup(3, 3.seconds)
+
+      projectionTestKit.run(projection2) {
+        projectedTestValueShouldBe("e1-1|e1-2|e1-3|e1-4|e1-5|e1-6")(pid1)
+        projectedTestValueShouldBe("e2-1|e2-2|e2-3|e2-4")(pid2)
+      }
+      latestOffsetShouldBe(envelopes2.last.offset)
+    }
+
+    "be able to skip envelopes but still store offset" in {
+      implicit val pid = UUID.randomUUID().toString
+      val projectionId = genRandomProjectionId()
+      val envelopes = createEnvelopes(pid, 6).map { env =>
+        if (env.event == "e3" || env.event == "e4" || env.event == "e6")
+          markAsFilteredEvent(env)
+        else
+          env
+      }
+      val sourceProvider = createSourceProvider(envelopes)
+      implicit val offsetStore = createOffsetStore(projectionId, sourceProvider)
+
+      val handlerProbe = createTestProbe[String]("calls-to-handler")
+
+      val projection =
+        DynamoDBProjection
+          .exactlyOnceGroupedWithin(
+            projectionId,
+            Some(settings),
+            sourceProvider,
+            handler = () => new TransactGroupedConcatHandler(handlerProbe.ref))
+          .withGroup(3, 3.seconds)
+
+      offsetShouldBeEmpty()
+      projectionTestKit.run(projection) {
+        projectedTestValueShouldBe("e1|e2|e5")
+      }
+
+      // handler probe is called twice
+      handlerProbe.expectMessage("called")
+      handlerProbe.expectMessage("called")
+
+      latestOffsetShouldBe(envelopes.last.offset)
+    }
+
+    "handle grouped async projection" in {
```

Comment:
Not important for test names, but still using the "grouped async" naming, rather than "at-least-once grouped".

```suggestion
    "handle at-least-once grouped projection" in {
```